src/MessagePack/Formatters/PrimitiveObjectFormatter.cs

```diff
@@ -37,7 +37,13 @@ protected PrimitiveObjectFormatter()
         {
         }
 
+        [Obsolete("Please, use the method overload without TypeInfo")]
         public static bool IsSupportedType(Type type, TypeInfo typeInfo, object value)
+        {
+            return IsSupportedType(type, value);
+        }
+
+        public static bool IsSupportedType(Type type, object value)
         {
             if (value == null)
```

Comment:
Should we use `if (value is null)` instead?

Reply:
Up to you. I prefer `is null` syntax, but it's not a big deal.

Reply:
Let's leave it for another PR

Reply:
You have that backwards. `is null` _never_ compiles against structs while `== null` sometimes will. 
[evidence](https://sharplab.io/#v2:C4LglgNgPgAgTARgLACgYGYAE9MGFMDeqmJ2WARgPaUSYCiAjgK4CGEAzgHJMQQAUASkwBeAHyYArCOGYAdjwgBuYqQyYqNTAEkuCwSPFSw7OQuUoAvkA===).

Reply:
In fact I wrote an analyzer to catch uses of `== null` and `!= null` and offer a code fix to switch to pattern matching:
https://github.com/AArnott/CSharpIsNull